packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -375,6 +406,13 @@ Future<void> _buildGradleProjectV2(
   String target,
   bool isBuildingBundle,
 ) async {
+  // TODO(mklim): This is breaking, should fix the incompatibility or only case
+  // into this for migrated apps (_buildGradleProjectV3)?
```

Comment:
I think we probably want this to be considered a new type of project, and we should offer some kind of auto-migration for it (e.g. removing any flutter/plugin related subproject nodes).